sdk/nodejs/provider/server.ts

```diff
@@ -37,14 +37,26 @@ class Server implements grpc.UntypedServiceImplementation {
     engineAddr: string | undefined;
     readonly provider: Provider;
     readonly uncaughtErrors: Set<Error>;
-
-    /** Queue of construct calls. */
-    constructCallQueue = Promise.resolve();
+    private readonly _callbacks = new Map<string, any>();
```

Comment:
Can we do better than `any` for the type here? It seems like they are all at least `(err: Error, ...) => void`, maybe?

Reply:
We could, but all the rpc methods are completely untyped: https://github.com/pulumi/pulumi/blame/master/sdk/nodejs/provider/server.ts#L381

Shouldn't we rather add types here first?